contrib/storage-ipfs/src/main/java/org/apache/drill/exec/store/ipfs/IPFSContext.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018-2020 Bowen Ding, Yuedong Xu, Liang Wang
```

Comment:
Please remove the custom license.

Reply:
We would like to retain a statement that reflects the plugin is contributed by these authors. Where would be a more appropriate place to put such a statement? Maybe `readme.md`?

Reply:
Authorship can be identified from the git history. To specify multiple authors, you may [create a commit with multiple authors](https://help.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors).

Over time this code may be changed by other committers, even only briefly and in minor ways, so it would be hard to keep the list of all the people, made contributions to the specific part of code

Reply:
@vvysotskyi 
Can @dbw9580 put the authors names and perhaps contact info in the `README`?  Personally, I don't have a problem with that.

Reply:
Neither current code nor docs don't have any mention about their authors, and I don't think that it should be changed. As I mentioned above, it would be hard to keep a list of **all** contributors for the specific code part. Git history allows to see specific contribution, but just putting the author name to the code doesn't show anything related to the changes. What would we do with authorship if the plugin will be rewritten in the future from scratch, or significantly refactored, so it would differ significantly from its initial version? Using just git history solves this issue, and I don't think that we should introduce more complexity.

Reply:
@vvysotskyi We don't intend to keep our names in an contributors list; we just need a proof that this plugin, at a certain state, was contributed by us. In fact, we started this as a research project, and I need such a proof to include in my master's thesis to show these works were really done by me. 

I understand that Git history is the best and reliable way to verify authorship; the problem is Git and Github could be too complicated for someone who is not familiar with Open Source software development, in contrast to how people give credits by simply citing literature in research papers.

If possible, I would like to add a commit just before merge, which adds a authorship statement in `README`, and then another commit to delete it. This way, I can use that commit as a proof. How does this sound to you?

Reply:
@dbw9580
Are you just looking for a permanent link for attribution?  If you go with the aforementioned approach, the issue is that we squash all commits into one commit prior to committing.  I suspect that your name change would be lost in the shuffle. 

Another option might be to put your names in both the JIRA and the pull request description.  You could also include a comment somewhere in the code pointing to the JIRA.  In publications I would assume you could then refer to the JIRA or pull request link.

@vvysotskyi would that be acceptable?

Reply:
> a permanent link for attribution

Yes.

> name change would be lost in the shuffle

Can we squash all the commits that are real code changes into one commit, followed by another two commits that simply add/delete the names?

Reply:
Yes, adding names to the Jira and PR description sounds good to me. By the way, Jira has {{Assignee}} filed. I've added your Jira account to the contributor role and assigned it's Jira to you.

Glad to hear that such contributions are made as a research project.

Reply:
> adding names to the Jira and PR description
Yeah we can accept that. Edited the PR description.